portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -53,16 +67,71 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
   /*
    * Converts List of comments into a JSON using the gson library.
    */
-  private String convertToJson(List<Comment> comments) {
+  private static String convertToJson(List<Comment> comments) {
     return new Gson().toJson(comments);
   }
 
   /**
    * @return the request parameter, or the default value if the parameter
    *         was not specified by the client
    */
-  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+  private static String getParameter(HttpServletRequest request, String name, String defaultValue) {
     String value = request.getParameter(name);
     return (value == null || value.isEmpty()) ? defaultValue : value;
   }
+
+  /** Returns number of comments entered by the user or -1 if choice is invalid */
+  private static int getNumberOfComments(HttpServletRequest request) {
+    // Get input from the form.
+    String numberOfCommentsString = request.getParameter("number-of-comments");
+
+    // Convert input to an int.
+    int numberOfComments;
+    try {
+      numberOfComments = Integer.parseInt(numberOfCommentsString);
+    } catch (NumberFormatException e) {
+      System.err.println("Cannot convert to int: " + numberOfCommentsString);
+      return -1;
```

Comment:
It is not recommended to use a illegal number to present failure.

I think you can either:
  1, Throw the Exception and catch it outside.
  2, Or make the function to return Optional<Integer>. So you can return Optional.empty() when you see failure.

Reply:
Done. 1st option taken - throw the exception and catch it outside (caught in the doGet Method and thrown in the getNumberOfComments method)